File: problems/problems_Interview_16__02/solution.go
package problemInterview_16__02

import (
	"encoding/json"
	"log"
	"strings"
)

type WordsFrequency struct {
    
}


func Constructor(book []string) WordsFrequency {
    
}


func (this *WordsFrequency) Get(word string) int {
    
}


/**
 * Your WordsFrequency object will be instantiated and called as such:
 * obj := Constructor(book);
 * param_1 := obj.Get(word);
 */

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var operators []string
	var opValues [][]any
	var ans []any
	if err := json.Unmarshal([]byte(inputValues[0]), &operators); err != nil {
		log.Println(err)
		return nil
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &opValues); err != nil {
		log.Println(err)
		return nil
	}
	var bookArr []string
	if v, ok := opValues[0][0].([]string); ok {
		bookArr = v
	} else {
		for _, vi := range opValues[0][0].([]any) {
			bookArr = append(bookArr, vi.(string))
		}
	}
	obj := Constructor(bookArr)
	ans = append(ans, nil)
	for i := 1; i < len(operators); i++ {
		var res any
		switch operators[i] {
		case "get", "Get":
			res = obj.Get(opValues[i][0].(string))
		default:
			res = nil
		}
		ans = append(ans, res)
	}


	return ans
}

File: problems/problems_Interview_16__02/solution.ts
class WordsFrequency {
    constructor(book: string[]) {
        
    }

    get(word: string): number {
        
    }
}

/**
 * Your WordsFrequency object will be instantiated and called as such:
 * var obj = new WordsFrequency(book)
 * var param_1 = obj.get(word)
 */

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const operators: string[] = JSON.parse(inputValues[0]);
	const opValues: any[][] = JSON.parse(inputValues[1]);
	const ans: any[] = [null];
	const obj: WordsFrequency = new WordsFrequency(opValues[0][0]);
	for (let i: number = 1; i < operators.length; i++) {
		if (operators[i] == "get") {
			ans.push(obj.get(opValues[i][0]));
			continue;
		}
		ans.push(null);
	}
	return ans;
}
